src/class/api.php

```diff
@@ -0,0 +1,356 @@
+                // We received version in URL.
+                // Version must be larger than zero, and no greater than the currently configured version.
+                if (!$aRegs[1] || $aRegs[1] > $this->nVersion) {
+                    $this->aResponse['errors'][] = 'Requested version is unavailable.';
+                    $this->sendHeader(400, true); // Send 400 Bad Request, print response, and quit.
+                } else {
+                    $this->nVersion = (int) $aRegs[1];
+                }
+                array_shift($aURLElements); // Take the version off.
+            }
+
+            // Next, should be resource.
+            $this->sResource = array_shift($aURLElements);
+            if (!isset($this->aResourcesSupported[$this->sResource])) {
+                $this->aResponse['errors'][] = 'Requested resource is unknown.';
+                $this->sendHeader(400, true); // Send 400 Bad Request, print response, and quit.
+            }
+
+            // From here, it's optional.
+            $this->nID = array_shift($aURLElements);
+
+            // Rest of the URL should be empty at this point.
+            if (implode('', $aURLElements)) {
+                // URL still had more data. At this point, that can't be right.
+                $this->aResponse['errors'][] = 'Could not parse requested URL.';
+                $this->sendHeader(400, true); // Send 400 Bad Request, print response, and quit.
+            }
+
+            // Verify method. This depends on the resource.
+            if (!in_array($_SERVER['REQUEST_METHOD'], $this->aResourcesSupported[$this->sResource])) {
+                $this->aResponse['errors'][] = 'Method not allowed here. Options: ' . implode(', ', $this->aResourcesSupported[$this->sResource]);
+                $this->sendHeader(405, true); // Send 405 Method Not Allowed, print response, and quit.
+            }
+
+            // The combination of POST with an ID can't work.
+            if (POST && $this->nID !== '') {
+                // Remove POST from options, before we mention it.
+                // FIXME: Yes, this currently means there are no methods left...
+                unset($this->aResourcesSupported[$this->sResource][array_search('POST', $this->aResourcesSupported[$this->sResource])]);
+                $this->aResponse['errors'][] = 'Method not allowed here. Options: ' . implode(', ', $this->aResourcesSupported[$this->sResource]);
+                $this->sendHeader(405, true); // Send 405 Method Not Allowed, print response, and quit.
+            }
+
+            // If we're here, the API regarded the call acceptable, and assigned
+            //  an API version higher than 0 (the LOVD2-style API).
+            // If we're at version 1 or higher, let this new API handle it.
+            // Since each method requires very specific code, the methods are
+            //  handled separately.
+            // These methods should process the request and stop PHP execution.
```

Comment:
You're right, it was speaking of the previous structure. I've removed the comment, thanks!